src/fheroes2/editor/editor_castle_details_window.cpp

```diff
@@ -588,11 +597,18 @@ namespace Editor
                     message = _( "Set custom Castle Army. Right-click to reset unit." );
                 }
             }
-            else if ( le.isMouseCursorPosInArea( buttonExit.area() ) ) {
-                message = _( "Exit Castle Options" );
+            else if ( le.isMouseCursorPosInArea( buttonOkay.area() ) ) {
+                message = _( "Click to accept the changes made." );
 
-                if ( le.isMouseRightButtonPressedInArea( buttonExit.area() ) ) {
-                    fheroes2::showStandardTextMessage( _( "Exit" ), message, Dialog::ZERO );
+                if ( le.isMouseRightButtonPressedInArea( buttonOkay.area() ) ) {
+                    fheroes2::showStandardTextMessage( _( "Okay" ), message, Dialog::ZERO );
+                }
+            }
+            else if ( le.isMouseCursorPosInArea( buttonCancel.area() ) ) {
+                message = _( "Exit this menu without doing anything." );
```

Comment:
Hi @ihhub and @zenseii, 
what do you think of the next rephrasing?
```suggestion
                message = _( "Exit this dialog, discarding the changes made." );
```